FWCore/Framework/test/productregistry.cppunit.cc

```diff
@@ -110,18 +110,18 @@ void testProductRegistry::setUp() {
       std::make_shared<edm::ProductDescription>(edm::InEvent,
                                                 "labelovsimple",
                                                 "PROD",
-                                                "edm::OwnVector<edmtest::Simple>",
+                                                edm::TypeID(typeid(edm::OwnVector<edmtest::Simple>)).userClassName(),
                                                 "edmtestSimplesOwned",
                                                 "ovsimple",
                                                 edm::TypeWithDict(typeid(edm::OwnVector<edmtest::Simple>)));
```

Comment:
I'm wondering why we pass the type information twice, once as a string (at least in
https://github.com/cms-sw/cmssw/blob/043863c8a6c31223348955184daba965c079af14/FWCore/Framework/src/ProductRegistryHelper.cc#L81
via `edm::TypeID`) and second time as `edm::TypeWithDict`.

I think we need to discuss what to do in the longer term (also in relation to https://github.com/cms-sw/framework-team/issues/436)